pkg/provider/aws/resources/rds.go

```diff
@@ -186,6 +186,78 @@ func (subnetGroup *RdsSubnetGroup) Create(dag *core.ResourceGraph, params RdsSub
 	return nil
 }
 
+type RdsProxyCreateParams struct {
+	AppName string
+	Name    string
+	Refs    []core.AnnotationKey
+}
+
+// Create takes in an all necessary parameters to generate the RdsProxy name and ensure that the RdsProxy is correlated to the constructs which required its creation.
+//
+// This method will also create dependent resources which are necessary for functionality. Those resources are:
+//   - Subnets
+//   - SecurityGroups
+func (proxy *RdsProxy) Create(dag *core.ResourceGraph, params RdsProxyCreateParams) error {
+	proxy.Name = rdsSubnetSanitizer.Apply(fmt.Sprintf("%s-%s", params.AppName, params.Name))
+	proxy.ConstructsRef = params.Refs
+
+	existingProxy := dag.GetResource(proxy.Id())
+	if existingProxy != nil {
+		graphProxy := existingProxy.(*RdsProxy)
+		graphProxy.ConstructsRef = core.DedupeAnnotationKeys(append(graphProxy.KlothoConstructRef(), params.Refs...))
+	} else {
+		proxy.Subnets = make([]*Subnet, 2)
```

Comment:
This is odd - can't this just be determined from the length of the `Subnets` params?

Reply:
I meant these:
```go
			"Subnets": []SubnetCreateParams{
				{
					AppName: params.AppName,
					Refs:    params.Refs,
					AZ:      "0",
					Type:    PrivateSubnet,
				},
				{
					AppName: params.AppName,
					Refs:    params.Refs,
					AZ:      "1",
					Type:    PrivateSubnet,
				},
			},
```

I was just wondering why create the array before calling the `CreateDependencies`